### ISSUE TITLE
High contrast broken when not logged in in /beta

### ISSUE THREAD

**Issue by jwunderl** [microsoft/pxt-microbit#5076]
**Describe the bug**

Can't enter high contrast mode in /beta when not logged in

**To Reproduce**
1. Go to /beta in incognito or browser you haven't logged in to (occurs in both microbit/beta and arcade/beta, but not arcade released site so recent regression)
2. Click on gearwheel in top right
3. try to turn on high contrast mode

**Expected behavior**

Should work

**Additional context**

throws this exception: 

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'preferences')
    at m.userPreferencesAsync (pxtapp.js:1:46084)
    at async m.patchUserPreferencesAsync (pxtapp.js:1:47104)
    at async Object.setHighContrastPrefAsync (main.js:1:205030)
    at async S (main.js:1:607186)
```
